### DISCUSSION TITLE
Soft labels in multiclassification settings.

### DISCUSSION THREAD

**Discussion by github-abdellah-elyounsi** [catboost/catboost#2964] [Q&A]
Hello,

I’m trying to train a CatBoost model for a multiclass problem using soft labels instead of hard labels.

My labels are vectors like [0.2, 0.7, 0.1] representing probabilities, instead of hard-coded class labels like [0, 1, 0].

Looking at the CatBoost documentation, it seems that for classification tasks (including multiclass), the labels are expected to be boolean, integer, or string values.

I’m wondering if CatBoost internally has a preprocessing step that automatically converts soft labels by mapping the maximum value to 1 and the rest to 0, or if it strictly requires hard labels.

If not, what would be the recommended way to train a model with soft labels using CatBoost?

Thanks in advance!

The code below works fine, but i'm afraid it's failing silently

```
from catboost import Pool, CatBoostClassifier

train_data = [["summer", 1924, 44],
              ["summer", 1932, 37],
              ["winter", 1980, 37],
              ["summer", 2012, 204]]

eval_data = [["winter", 1996, 197],
             ["winter", 1968, 37],
             ["summer", 2002, 77],
             ["summer", 1948, 59]]

cat_features = [0]

train_label = [0.05, 0.8, 0.05, 0.1]
eval_label = [0.05, 0.8, 0.05, 0.1]


train_dataset = Pool(data=train_data,
                     label=train_label,
                     cat_features=cat_features)

eval_dataset = Pool(data=eval_data,
                    label=eval_label,
                    cat_features=cat_features)

model = CatBoostClassifier(iterations=10,
                           learning_rate=1,
                           depth=2,
                           loss_function='MultiClass')

model.fit(train_dataset)

preds_class = model.predict(eval_dataset)

preds_proba = model.predict_proba(eval_dataset)

preds_raw = model.predict(eval_dataset,
                          prediction_type='RawFormulaVal')
```
